### PR TITLE
Consistent go to user

### PR DESCRIPTION
**Pull request by howard-stearns**
Make all PAL "go to user" paths ultimately go through the same code.
As a bonus. make that code work when you to to user who is in the sky.
Fixes https://highfidelity.fogbugz.com/f/cases/8341

### CODE REVIEW

**Comment by howard-stearns** [comment]
**test plan:**
NOTE: You need two machines and domain that works correctly with multiple users. Probably best to wait until that's fixed, and I update this PR by merging in the solution. (I developed the solution by including https://github.com/highfidelity/hifi/pull/11995, but since that's not in master, it is not in this PR yet.)

1. Test going to a user in both the the "nearby" and "connections" tab of the PEOPLE app. It should work reliably, regardless of whether the person you are going to is on the ground or in the air.

2. Run the navigation test suite S256 (e.g., including "Can't get stuck in surfaces or floors", "Flying should work", etc.)  Of course, the results of this PR should match that of master - no promises of doing better than master!

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5681/app_12002_PR.apk

File: interface/resources/qml/hifi/NameCard.qml [hunk 1]
```diff
@@ -597,18 +597,11 @@ Item {
...
-        // the avatar goes into fly mode rather than falling. However, that is not exposed to Javascript right now.
-        // FIXME: it would be nice if this used the same teleport steps and smoothing as in the teleport.js script.
-        // Note, however, that this script allows teleporting to a person in the air, while teleport.js is going to a grounded target.
-        // Position avatar 2 metres from the target in the direction that target avatar was facing.
-        MyAvatar.position = Vec3.sum(avatar.position, Vec3.multiplyQbyV(avatar.orientation, {x: 0, y: 0, z: -2}));
-        
-        // Rotate avatar on Y axis to face target avatar and cancel out any inherited roll and pitch.
-        MyAvatar.orientation = Quat.cancelOutRollAndPitch(Quat.multiply(avatar.orientation, {y: 1}));
+        // This is the last step of what AddressManager.goToUser does, but we don't need to resolve the username.
+        MyAvatar.goToLocation(avatar.position, true, Quat.cancelOutRollAndPitch(avatar.orientation), true)
```
Comment (howard-stearns) [inline interface/resources/qml/hifi/NameCard.qml]: This is the main thing: going through the code that goToUser ultimately goes through, per the comment.

File: interface/src/avatar/MyAvatar.cpp [hunk 2]
```diff
@@ -424,6 +424,7 @@ void MyAvatar::update(float deltaTime) {
         emit positionGoneTo();
         // Run safety tests as soon as we can after goToLocation, or clear if we're not colliding.
         _physicsSafetyPending = getCollisionsEnabled();
+        _characterController.recomputeFlying(); // In case we've gone to into the sky.
```
Comment (howard-stearns) [inline interface/src/avatar/MyAvatar.cpp]: When we go somewhere, ask the characterController to recomputeFlying when it updates.

File: libraries/physics/src/CharacterController.cpp [hunk 3]
```diff
@@ -391,6 +391,10 @@ void CharacterController::setState(State desiredState) {
     }
 }
 
+void CharacterController::recomputeFlying() {
+    _pendingFlags |= PENDING_FLAG_RECOMPUTE_FLYING;
+}
+
```
Comment (howard-stearns) [inline libraries/physics/src/CharacterController.cpp]: We don't actually do anything just yet.... See below.

File: libraries/physics/src/CharacterController.cpp [hunk 4]
```diff
@@ -657,6 +661,13 @@ void CharacterController::updateState() {
     if (!_dynamicsWorld) {
         return;
     }
+    if (_pendingFlags & PENDING_FLAG_RECOMPUTE_FLYING) {
+        SET_STATE(CharacterController::State::Hover, "recomputeFlying");
+         _hasSupport = false;
+         _stepHeight = _minStepHeight; // clears memory of last step obstacle
+         _pendingFlags &= ~PENDING_FLAG_RECOMPUTE_FLYING;
+    }
+
```
Comment (howard-stearns) [inline libraries/physics/src/CharacterController.cpp]: This addition to updateState resets things so that we can properly tell if we're flying.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5681/HighFidelity-Beta-PR12002-813a94a79ed5f803aa870f4642eee00d95187cdb.dmg

**Comment by ZappoMan** [comment]
gustavo test this please

File: interface/resources/qml/hifi/NameCard.qml (hunk 1 above)
Comment (ZappoMan) [inline interface/resources/qml/hifi/NameCard.qml]: can we get a trailing semicolon please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5684/app_12002_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5684/HighFidelity-Beta-PR12002-813a94a79ed5f803aa870f4642eee00d95187cdb.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5684/HighFidelity-Beta-PR12002-813a94a79ed5f803aa870f4642eee00d95187cdb.dmg

File: interface/resources/qml/hifi/NameCard.qml (hunk 1 above)
Comment (howard-stearns) [reply interface/resources/qml/hifi/NameCard.qml]: doh! of course.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5725/app_12002_PR.apk

**Comment by howard-stearns** [comment]
This now pulls in https://github.com/highfidelity/hifi/pull/11995 from master, so it should be testable as soon as it builds.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5726/app_12002_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5726/HighFidelity-Beta-PR12002-bb499000a6ad7f19dfe232e22f6a3f6f0dcce8ab.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5755/app_12002_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5755/HighFidelity-Beta-PR12002-bb499000a6ad7f19dfe232e22f6a3f6f0dcce8ab.dmg

**Comment by sbennett77** [comment]
Currently Testing.

**Comment by sbennett77** [comment]
Testing passed for test plan on this page, as well as S256.


@conklin94122

**Pull request merged**